diff --git a/entityreference/behavior/UuidEntityReferenceBehavior.class.php b/entityreference/behavior/UuidEntityReferenceBehavior.class.php
index 9d9aa6a..b8aed82 100755
--- a/entityreference/behavior/UuidEntityReferenceBehavior.class.php
+++ b/entityreference/behavior/UuidEntityReferenceBehavior.class.php
@@ -77,8 +77,17 @@ class UuidEntityReferenceBehavior extends EntityReference_BehaviorHandler_Abstra
 
     foreach ($items as &$item) {
       if (!empty($item['target_id']) && empty($item['uuid'])) {
-        $wrapper = entity_metadata_wrapper($target_type, $item['target_id']);
-        $item['uuid'] = $wrapper->{$uuid_field_name}->value();
+        try {
+          $wrapper = entity_metadata_wrapper($target_type, $item['target_id']);
+          $item['uuid'] = $wrapper->{$uuid_field_name}->value();
+        }
+        catch (Exception $e) {
+          $entity = entity_load_unchanged($target_type, $item['target_id']);
+          if (empty($entity->uuid)) {
+            throw $e;
+          }
+          $item['uuid'] = $entity->uuid;
+        }
       }
     }
   }
